CustomerRegister/Services/CustomerService.cs

```diff
@@ -0,0 +1,74 @@
+using CustomerRegister.Repositories.Interfaces;
+using Microsoft.AspNetCore.Http.HttpResults;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CustomerRegister
+{
+    public class CustomerService : ICustomerService
+    {
+        private readonly List<CustomerEntity> _customers = new();
+
+        public bool ExistsID(CustomerEntity selectedCustomer)
+        {
+            var response = _customers
+                .Any(customer => customer.Id != selectedCustomer.Id);
+            return response;
+        }
+
+        public void DuplicatedEmail(CustomerEntity selectedCustomer)
+        {
+            if(_customers.Any(customer => customer.Email == selectedCustomer.Email))
+                throw new ArgumentException("Este email já está em uso, por favor escolha outro");
+        }
+
+        public void DuplicatedCPF(CustomerEntity selectedCustomer)
+        {
+            if (_customers.Any(customer => customer.Cpf == selectedCustomer.Cpf))
+                throw new ArgumentException("Este CPF já está em uso, por favor escolha outro");
+        }
```

Comment:
Esses dois métodos são utilizados para verificar se um customer já existe. Você poderia criar um único e colocar os dois condicionai dentro deste. Assim, quando fosse criar ou atualizar, ele chamaria esse método e verificaria.
```suggestion
        private void CustomerAlreadyExists(CustomerEntity selectedCustomer)
        {
            if(_customers.Any(customer => customer.Email == selectedCustomer.Email))
                throw new ArgumentException("Este email já está em uso, por favor escolha outro");
                
            if (_customers.Any(customer => customer.Cpf == selectedCustomer.Cpf))
                throw new ArgumentException("Este CPF já está em uso, por favor escolha outro");
        }
```